pkg/recommendation/factory_test.go

```diff
@@ -0,0 +1,76 @@
+package recommendation_test
+
+import (
+	"testing"
+
+	logtest "github.com/sirupsen/logrus/hooks/test"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+
+	"github.com/kubeshop/botkube/pkg/config"
+	"github.com/kubeshop/botkube/pkg/recommendation"
+)
+
+func TestFactory_NewForSources(t *testing.T) {
+	// given
+	sources := config.IndexableMap[config.Sources]{
```

Comment:
we should stop using the `config.IndexableMap` in places where `GetFirst()` is not needed.